cmd/siad/server.go

```diff
@@ -366,6 +368,21 @@ func (srv *Server) daemonHandler(password string) http.Handler {
 	return router
 }
 
+// uninitializedHandler handles calls to routes that have not yet been
+// initialized.
+func uninitializedHandler(w http.ResponseWriter, r *http.Request) {
+	api.WriteError(w, api.Error{"siad is not ready. please wait for siad to finish loading."}, http.StatusServiceUnavailable)
+}
+
+// AttachAPI attaches an initialized Sia API to the server.
+func (srv *Server) AttachAPI(a *api.API) {
+	mux := http.NewServeMux()
+	mux.Handle("/daemon/", api.RequireUserAgent(srv.daemonHandler(srv.requiredPassword), srv.requiredUserAgent))
+	mux.Handle("/", a)
+	srv.mux = mux
+	srv.httpServer.Handler = mux
```

Comment:
isn't this racey?

Reply:
It may be. I'll write a test for this behavior that uses the race detector